File: 001~100/048.go
// URL : https://leetcode.com/problems/rotate-image/
//0 ms	2.7 MB
package main
import "fmt"

//----------------------------------- Answer Begin -----------------------------------//
func rotate(matrix [][]int){
    size := len(matrix)
    for i:=0; i<size/2; i++{
        for j:=i; j<size-i-1; j++{
            matrix[i][j], matrix[j][size-1-i], matrix[size-1-i][size-1-j], matrix[size-1-j][i] = matrix[size-1-j][i], matrix[i][j], matrix[j][size-1-i], matrix[size-1-i][size-1-j]
        }
    }
}
//----------------------------------- Answer END -----------------------------------//

func printM(arr [][]int){
    for i:=0; i<len(arr); i++{
        for j:=0; j<len(arr); j++{
            fmt.Printf("%2d ",arr[i][j])
        }
        fmt.Println()
    }
    fmt.Println()
}

func main(){
    arr:=[][]int{
        {1},
    }
    fmt.Println("before:")
    printM(arr)
    rotate(arr)
    fmt.Println("after:")
    printM(arr)

    arr2:=[][]int{
        {1,2},
        {3,4},
    }
    fmt.Println("before:")
    printM(arr2)
    rotate(arr2)
    fmt.Println("after:")
    printM(arr2)

    arr3:=[][]int{
        {1,2,3},
        {4,5,6},
        {7,8,9},
    }
    fmt.Println("before:")
    printM(arr3)
    rotate(arr3)
    fmt.Println("after:")
    printM(arr3)

    arr4:=[][]int{
        {1,2,3,4},
        {5,6,7,8},
        {9,10,11,12},
        {13,14,15,16},
    }
    fmt.Println("before:")
    printM(arr4)
    rotate(arr4)
    fmt.Println("after:")
    printM(arr4)
}

File: 001~100/048.php
<?php
// URL : https://leetcode.com/problems/rotate-image/
//8 ms	14.9 MB
class Solution {
    public $matrix;
    function rotate(&$matrix) {
        $size=count($matrix);
        for($i=0; $i<(int)($size/2); $i++){
            for($j=$i; $j<$size-$i-1; $j++){
                $cache=$matrix[$i][$j];
                $matrix[$i][$j]=$matrix[$size-1-$j][$i];
                $matrix[$size-1-$j][$i]=$matrix[$size-1-$i][$size-1-$j];
                $matrix[$size-1-$i][$size-1-$j]=$matrix[$j][$size-1-$i];
                $matrix[$j][$size-1-$i]=$cache;
            }
        }
    }
}

function printM(&$arr){
    for($i=0; $i<count($arr); $i++){
        for($j=0; $j<count($arr); $j++){
            echo $arr[$i][$j]." ";
        }
        echo '<br>';
    }
    echo '<br>';
}

$sol=new Solution;

$arr=Array(
        Array(1)
    );
$sol->matrix=$arr;
$sol->rotate($sol->matrix);
printM($sol->matrix);

$arr=Array(
        Array(1,2),
        Array(3,4),
    );
$sol->matrix=$arr;
$sol->rotate($sol->matrix);
printM($sol->matrix);

$arr=Array(
        Array(1,2,3),
        Array(4,5,6),
        Array(7,8,9)
    );
$sol->matrix=$arr;
$sol->rotate($sol->matrix);
printM($sol->matrix);

$arr=Array(
        Array(1,2,3,4),
        Array(5,6,7,8),
        Array(9,10,11,12),
        Array(13,14,15,16)
    );
$sol->matrix=$arr;
$sol->rotate($sol->matrix);
printM($sol->matrix);
?>
